Gemini/src/UrlMinter/UrlMinter.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: daniel
+ * Date: 19/06/17
+ * Time: 2:13 PM
+ */
+
+namespace Islandora\Gemini\UrlMinter;
+
+class UrlMinter implements UrlMinterInterface
+{
+    protected $base_url;
+
+    public function __construct(
+        $base_url
+    ) {
+        $trimmed = trim($base_url);
+        $this->base_url = rtrim($trimmed, '/') . '/';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function mint($context)
+    {
+        if (strlen($context) < 8) {
+            throw new \InvalidArgumentException(
+                "Provided UUID must be at least of length 8 to account for pair-trees",
+                400
+            );
+        }
+
+        $segments = [
+            substr($context, 0, 2),
+            substr($context, 2, 2),
+            substr($context, 4, 2),
+            substr($context, 6, 2),
+        ];
```

Comment:
I think this could be reduced to
`$segments = str_split($context, 2);`
http://php.net/str_split

Reply:
@whikloj++